src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -28,20 +29,19 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
-  public boolean fire(int numberOfTorpedos){
-    if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+  public boolean fire(int numberOfTorpedos) {
+    if (numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount) {
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
```

Comment:
This line would also change according to the privous
```suggestion
    double r = randomGenerator.nextDouble();
```